bemani/data/cache.py

```diff
@@ -0,0 +1,53 @@
+from typing import Any, Callable
+from functools import wraps
+import hashlib
+import pickle
+# Make memcache optional
+try:
+    import pylibmc  # type: ignore
+    has_mc = True
+except ModuleNotFoundError:
+    has_mc = False
+
+
+def cached(lifetime: int=10, extra_key: Any=None) -> Callable:
+    def _cached(func: Callable) -> Callable:
+        if has_mc:
+            memcache = pylibmc.Client(["127.0.0.1"], binary=True)
+            memcache.behaviors = {"tcp_nodelay": True, "ketama": True}
+        else:
+            memcache = None
+
+        @wraps(func)
+        def wrapper(*args: Any, **kwargs: Any) -> Any:
+            if memcache is None:
```

Comment:
Do this outside the wrapper so it doesn't have to be checked every call. Just return a different constructed function that wraps without affecting anything.